### PR TITLE
feat(ckb-bin): add interactive mode for init sub command

### LINKED ISSUE #1039: `ckb init` interactive mode
Add an option `-i | --interactive` to enable the interactive mode. In this mode, CKB will

- Ask whether to overwrite the config files.
- Check the database, if the database is incompatible, ask whether to clean the database directory.
- Ask whether to set up the block assembler. If yes, forward with a wizard to set up block assembler.

### PR DESCRIPTION
**Pull request by zjhmale**
Closes #1039

### CODE REVIEW

File: ckb-bin/src/subcommand/init.rs [hunk 1]
```diff
@@ -8,14 +14,108 @@ use ckb_script::Runner;
...
+        eprintln!("Config files already exists, use --force to overwrite.");
+
+        if args.interactive {
+            write!(stdout, "Overwrite config file now?").unwrap();
+            stdout.flush().unwrap();
+
+            let mut input = String::new();
+            let _ = stdin.read_line(&mut input);
+
+            if input != "y\n" {
```
Comment (keroro520) [inline ckb-bin/src/subcommand/init.rs]: Trim spaces and check `Y`.

```
["y", "Y"].contains(&input.trim())
```

Comment (zjhmale) [reply ckb-bin/src/subcommand/init.rs]: yeah, sure

File: ckb-bin/src/subcommand/init.rs [hunk 2]
```diff
@@ -8,14 +14,108 @@ use ckb_script::Runner;
...
         }
         return Ok(());
     }
 
+    let exported = Resource::exported_in(&args.root_dir);
+    if !args.force && exported {
+        eprintln!("Config files already exists, use --force to overwrite.");
+
+        if args.interactive {
+            write!(stdout, "Overwrite config file now?").unwrap();
```
Comment (quake) [inline ckb-bin/src/subcommand/init.rs]: Please extract all stdin/stdout related code fragment into a helper `prompt` function: 

```
    let input: String = prompt("Overwrite config file now?");
    if !["y", "Y"].contains(&input) {
        return Err(ExitCode::Failure);
    }   
```

Comment (zjhmale) [reply ckb-bin/src/subcommand/init.rs]: sure :) that will be more concise

Comment (zjhmale) [reply ckb-bin/src/subcommand/init.rs]: @quake done, pls review again :)

**Pull request merged**